src/generator/AutoRest.Ruby.Azure/TemplateModels/AzureMethodTemplateModel.cs

```diff
@@ -273,7 +273,8 @@ public override string OperationReturnTypeString
         {
             get
             {
-                if (Extensions.ContainsKey("nextMethodName") && !Extensions.ContainsKey(AzureExtensions.PageableExtension))
+                if (Extensions.ContainsKey("nextMethodName") && (!Extensions.ContainsKey(AzureExtensions.PageableExtension) ||
+                    (Extensions.ContainsKey(AzureExtensions.PageableExtension) && Extensions.ContainsKey(AzureExtensions.LongRunningExtension))))
```

Comment:
looks like we don't need this `(!Extensions.ContainsKey(AzureExtensions.PageableExtension) || (Extensions.ContainsKey(AzureExtensions.PageableExtension)`